src/Base/Matrix.cpp

```diff
@@ -23,15 +23,14 @@
 
 #include "PreCompiled.h"
 #ifndef _PreComp_
-# include <memory>
-# include <cstring>
 # include <sstream>
+# include <string>

```

Comment:
Why do you add string when it's already included in the header?

Reply:
Because I got a compilation error. I found out yesterday, that the MSVC compiler use different files, not necessarily the one in the header, see my next comment.

Reply:
Yes.